entity-server-api/src/main/java/org/terracotta/entity/ConcurrencyStrategy.java

```diff
@@ -38,20 +41,24 @@
   public final int UNIVERSAL_KEY = Integer.MIN_VALUE; 
```

Comment:
Depends on what you mean by "shortest path".  While I am not sure how aggressive the current implementation is, the plan is that any message scheduled with this key will be enqueued in the shortest execution queue.  This is possible since it provides no ordering guarantees.

Whether or not this is a good idea for EhCache will depend entirely on the way per-key data is stored and accessed.  If concurrent access is possible, then this might be worth looking at.  If the storage is optimized for single-threaded access, then you still need to schedule it in the appropriate key.

We probably want to get collect some data regarding the execution queue depth before trying to use this optimization in a situation where there would be any trade-off.  If the queues stay short or are evenly distributed (in a typical use-case), then this wouldn't do much to help.